prov/sockets/src/sock_progress.c

```diff
@@ -2154,23 +2154,20 @@ static int sock_pe_progress_rx_pe_entry(struct sock_pe *pe,
 					struct sock_rx_ctx *rx_ctx)
 {
 	int ret;
-	struct sock_conn *conn;
 
 	if (sock_comm_is_disconnected(pe_entry)) {
 		SOCK_LOG_DBG("conn disconnected: removing fd from pollset\n");
-		sock_pe_poll_del(pe, pe_entry->conn->sock_fd);
-		ofi_close_socket(pe_entry->conn->sock_fd);
-
-		fastlock_acquire(&pe_entry->ep_attr->lock);
-		sock_epoll_del(&pe_entry->ep_attr->cmap.epoll_set,
-			       pe_entry->conn->sock_fd);
-		conn = idm_lookup(&pe_entry->ep_attr->conn_idm, pe_entry->conn->sock_fd);
-		if (conn) {
-			sock_conn_reset_entry(conn);
-			idm_clear(&pe_entry->ep_attr->conn_idm, pe_entry->conn->sock_fd);
+		if (!pe_entry->conn->released) {
+			pe_entry->conn->released = 1;
+			fastlock_acquire(&pe_entry->ep_attr->cmap.lock);
+			sock_pe_poll_del(pe, pe_entry->conn->sock_fd);
+			sock_epoll_del(&pe_entry->ep_attr->cmap.epoll_set,
+					pe_entry->conn->sock_fd);
+			pe_entry->ep_attr->cmap.used--;
+			ofi_close_socket(pe_entry->conn->sock_fd);
+			fastlock_release(&pe_entry->ep_attr->cmap.lock);
```

Comment:
Refactored common conn_map release code to <code>sock_conn_release_entry</code> in sock_conn.c